src/main/java/com/mongodb/kafka/connect/sink/MongoSinkTopicConfig.java

```diff
@@ -513,12 +513,11 @@ public NamespaceMapper getNamespaceMapper() {
         .validateAll(sinkTopicOriginals)
         .forEach(
             (k, v) -> {
-              if (!v.errorMessages().isEmpty()) {
-                containsError.set(true);
-              }
-
               String name = topicOverrides.contains(prefix + k) ? prefix + k : k;
               if (props.containsKey(name) && !IGNORED_CONFIGS.contains(name)) {
+                if (!v.errorMessages().isEmpty()) {
+                  containsError.set(true);
```

Comment:
[this comment does not propose a change and does not prevent merging the PR]
The fact that we need to abuse `AtomicBoolean` as a mutable boolean holder suggests that for-each loop may be a better fit here than the `forEach` method.

Reply:
Ack